ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -123,8 +129,9 @@ void submitGetBlobOperation(String blobId, FutureResult<ReadableStreamChannel> f
     try {
       GetBlobOperation getBlobOperation =
           new GetBlobOperation(routerConfig, routerMetrics, clusterMap, responseHandler, blobId, futureResult, callback,
-              operationCompleteCallback, blobIdFactory, time);
+              operationCompleteCallback, readyForPollCallback, blobIdFactory, time);
       getOperations.add(getBlobOperation);
+      readyForPollCallback.onPollReady();
```

Comment:
I would prefer doing all the `onPollReady()` on operation arrival in `OperationController` and passing in the `readyForPollCallback` only when required (i.e. `GetBlobOperation` and `ChunkFiller`).